### PR TITLE
Feature/contact stent 2

### PR DESCRIPTION
**Pull request by BenceVirtonomy**

### CODE REVIEW

File: SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp [hunk 1]
```diff
@@ -124,8 +124,10 @@ StructuralSimulationInput::StructuralSimulationInput(
...
 	material_model_list_(material_model_list),
 	physical_viscosity_(physical_viscosity),
-	contacting_bodies_list_(contacting_bodies_list)
+	contacting_body_pairs_list_(contacting_bodies_list)
 {
+	//time dependent contact
+	time_dep_contacting_body_pairs_list_ = {};
 	// particle_relaxation option
 	particle_relaxation_list_ = {};
 	for (unsigned i = 0; i < resolution_list_.size(); i++){ particle_relaxation_list_.push_back(true); }
```
Comment (Xiangyu-Hu) [inline SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp:133]: Also size_t?

File: SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp [hunk 2]
```diff
@@ -221,7 +224,7 @@ void StructuralSimulation::ScaleTranslationAndResolution()
 void StructuralSimulation::SetSystemResolutionMax()
```
Comment (Xiangyu-Hu) [inline SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp:224]: for function name, using minor first letter of the verb?

File: cases_high_level_simulation/test_3d_unit_position_based_bc.cpp/CMakeLists.txt [hunk 3]
```diff
@@ -20,16 +20,15 @@ foreach(STL_FILE ${FILES_STL})
 endforeach()
 
 aux_source_directory(. DIR_SRCS)
-ADD_EXECUTABLE(${PROJECT_NAME} ${EXECUTABLE_OUTPUT_PATH}  ${DIR_SRCS})
```
Comment (Xiangyu-Hu) [inline cases_high_level_simulation/test_3d_unit_position_based_bc.cpp/CMakeLists.txt:23]: These changes are required for running the test in windows system.

File: cases_high_level_simulation/test_3d_unit_time_dep_contact/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+aux_source_directory(. DIR_SRCS)
+ADD_EXECUTABLE(${PROJECT_NAME} ${DIR_SRCS})
+
+add_test(NAME ${PROJECT_NAME}_particle_relaxation 
+		 COMMAND ${PROJECT_NAME} --r=true
+		 WORKING_DIRECTORY ${EXECUTABLE_OUTPUT_PATH})
+
+if(NOT SPH_ONLY_STATIC_BUILD) # usual dynamic build
+	if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
```
Comment (Xiangyu-Hu) [inline cases_high_level_simulation/test_3d_unit_time_dep_contact/CMakeLists.txt:36]: need to run the test in Windows system.

**Review by Xiangyu-Hu — APPROVED** [review]

File: SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp (hunk 2 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp]: done

File: SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp (hunk 1 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp]: done

File: cases_high_level_simulation/test_3d_unit_position_based_bc.cpp/CMakeLists.txt (hunk 3 above)
Comment (BenceVirtonomy) [reply cases_high_level_simulation/test_3d_unit_position_based_bc.cpp/CMakeLists.txt]: I put it back

File: cases_high_level_simulation/test_3d_unit_time_dep_contact/CMakeLists.txt (hunk 4 above)
Comment (BenceVirtonomy) [reply cases_high_level_simulation/test_3d_unit_time_dep_contact/CMakeLists.txt:36]: what do you mean here?

Comment (chuvirtonomy) [reply cases_high_level_simulation/test_3d_unit_time_dep_contact/CMakeLists.txt:36]: @Xiangyu-Hu we should make a pipeline for windows

Comment (chuvirtonomy) [reply cases_high_level_simulation/test_3d_unit_time_dep_contact/CMakeLists.txt:36]: on: push
jobs:
  test:
    strategy:
      matrix:
        platform: [ubuntu-latest, macos-latest, windows-latest]
    runs-on: ${{ matrix.platform }}
    steps:
    - uses: actions/checkout@v1
    - uses: actions/setup-node@v1
      with:
        version: 12
    - run: npm install-ci-test
    - uses:










It looks like this:

  publish:
    needs: [build]
    steps:
    - uses: actions/checkout@v1

**Pull request merged**